src/consensus.rs

```diff
@@ -0,0 +1,125 @@
+use std::{
+    sync::mpsc::{self, Receiver, RecvTimeoutError, Sender},
+    time::{Duration, Instant},
+};
+
+use raft::{prelude::*, storage::MemStorage};
+
+pub struct Consensus {
+    node: RawNode<MemStorage>,
+    receiver: Receiver<Message>,
+}
+
+impl Consensus {
+    pub fn new(logger: &slog::Logger) -> (Self, Sender<Message>) {
+        let config = Config {
+            id: 1,
+            ..Default::default()
+        };
+        config.validate().unwrap();
+        let storage = MemStorage::new_with_conf_state((vec![1], vec![]));
+        let node = RawNode::new(&config, storage, logger).unwrap();
+        let (sender, receiver) = mpsc::channel();
+        (Self { node, receiver }, sender)
+    }
+
+    pub fn start(&mut self) {
+        let mut t = Instant::now();
+        let mut timeout = Duration::from_millis(100);
+
+        loop {
+            match self.receiver.recv_timeout(timeout) {
+                Ok(message) => self.node.step(message).unwrap(),
+                Err(RecvTimeoutError::Timeout) => (),
+                Err(RecvTimeoutError::Disconnected) => return,
+            }
+
+            let d = t.elapsed();
+            t = Instant::now();
+            if d >= timeout {
+                timeout = Duration::from_millis(100);
```

Comment:
maybe the Raft ticking period could be a configuration knob?

Reply:
Let's move it to a const for now? There will be a lot of further work on consensus but I guess in later PRs.

Reply:
Good point, I am fine with a `const` for now then :+1: